signing_test.go

```diff
@@ -0,0 +1,77 @@
+package matrixfederation
+
+import (
+	"bytes"
+	"encoding/json"
+	"golang.org/x/crypto/ed25519"
+	"testing"
+)
+
+func TestSignJSON(t *testing.T) {
+	random := bytes.NewBuffer([]byte("Some 32 randomly generated bytes"))
+	entityName := "example.com"
+	keyID := "ed25519:my_key_id"
+	input := []byte(`{"this":"is","my":"message"}`)
+
+	publicKey, privateKey, err := ed25519.GenerateKey(random)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signed, err := SignJSON(entityName, keyID, privateKey, input)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = VerifyJSON(entityName, keyID, publicKey, signed)
```

Comment:
At the moment you're only really testing that stuff *you* sign will pass your verification process. It isn't testing that the signed JSON is anything other people can understand. Leo made some examples in [the appendices of the spec](https://matrix.org/docs/spec/appendices.html) - It would be nice if you could write a test which used that data and referenced it.